tower-balance/src/load/peak_ewma.rs

```diff
@@ -0,0 +1,212 @@
+use futures::{Async, Poll};
+use std::marker::PhantomData;
+use std::sync::{Arc, Mutex};
+use std::time::{Duration, Instant};
+use tower_discover::{Change, Discover};
+use tower_service::Service;
+
+use super::{Measure, MeasureFuture, NoMeasure};
+
+use Load;
+
+pub struct WithPeakEWMA<D, M> {
+    discover: D,
+    decay: Duration,
+    _p: PhantomData<M>,
+}
+
+pub struct PeakEWMA<S, M> {
+    service: S,
+    node: Arc<Mutex<Node>>,
+    _p: PhantomData<M>,
+}
+
+pub struct Instrument {
+    start: Instant,
+    node: Arc<Mutex<Node>>,
+}
+
+struct Node {
+    last_update: Instant,
+    rtt_estimate: f64,
+    tau: f64,
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, PartialOrd)]
+pub struct Cost(f64);
+
+// ===== impl PeakEWMA =====
+
+impl<D: Discover> WithPeakEWMA<D, NoMeasure> {
+    pub fn new(discover: D, decay: Duration) -> Self {
+        Self {
+            discover,
+            decay,
+            _p: PhantomData,
+        }
+    }
+
+    pub fn measured<M>(self) -> WithPeakEWMA<D, M>
```

Comment:
Should this be a builder? The main risk is that it could cause some forwards compat issues.

Reply:
@carllerche I'm open to this (I don't really understand the tradeoffs), but would opt to do that in a followup.